### PR TITLE
updated tables and column

### PR DESCRIPTION
**Pull request by merlouduron**
# Pull Request Review Comments

## Overview
- **Purpose of the PR**:
    - [ ] Feature addition
    - [ ] Bug fix
    - [ ] Refactoring
    - [ ] Documentation update
    - [ ] Other (please specify below)

## Specific Areas for Verification
- Please specify any particular requests or questions for the reviewer here.

## Suggestions/Concerns
- If you have any suggestions or concerns, please mention them here.

---

_**Additional Comments for the Reviewer (Optional)**_

- Add any additional comments or information for the reviewer here.

### CODE REVIEW

File: backend/api/models.py [hunk 1]
```diff
@@ -127,52 +119,56 @@ def save(self, *args, **kwargs):
 
 
 class PlanningProjectData(models.Model):
```
Comment (edward-stalley-work) [inline backend/api/models.py:121]: Will be updated to  `class Projects(models.Model):`

File: backend/api/models.py [hunk 2]
```diff
@@ -127,52 +119,56 @@ def save(self, *args, **kwargs):
...
+    project_id = models.CharField(
         max_length=10, primary_key=True, editable=False
     )
-    planning_project_name = models.CharField(max_length=100)
-    planning_project_type = models.CharField(max_length=50, null=True)
+    project_name = models.CharField(max_length=100)
+    project_type = models.CharField(max_length=50, null=True)
     client_id = models.ForeignKey(
         "ClientMaster", on_delete=models.CASCADE, related_name="planning_project"
     )
```
Comment (edward-stalley-work) [inline backend/api/models.py:129]: Currently in the DB the column name is client_id_id.
Lets try to make sure we just have one id. eg. 'client_id' NOT 'client_id_id'!

A few other columns in other tables in the DB also have this double '_id_id' issue.
Eg.
users.affiliated_company_id_id (original name)

File: backend/api/models.py [hunk 3]
```diff
@@ -127,52 +119,56 @@ def save(self, *args, **kwargs):
...
     planning = models.DateField(null=True, default=timezone.now)
```
Comment (edward-stalley-work) [inline backend/api/models.py:130]: What is this? Can it be removed.
Please double check if it is in code and let me know if it has a purpose.

File: backend/api/models.py [hunk 4]
```diff
@@ -239,34 +264,38 @@ def save(self, *args, **kwargs):
...
+    updated_at = models.DateTimeField(auto_now_add=True)
+    class Meta :
+        db_table = u'employee_expenses'
 
     def __str__(self):
         return self.planning_assign_id
     
 class CostOfSales(models.Model):
-    id = models.CharField(max_length=10, primary_key=True)
+    cost_of_sales_id = models.CharField(max_length=10, primary_key=True)
```
Comment (edward-stalley-work) [inline backend/api/models.py:285]: cost_of_sale_id

File: backend/api/models.py [hunk 5]
```diff
@@ -283,20 +312,24 @@ def __str__(self):
         return self.id
     
 class Expenses(models.Model):
-    id = models.CharField(max_length=10, primary_key=True)
+    expenses_id = models.CharField(max_length=10, primary_key=True)
```
Comment (edward-stalley-work) [inline backend/api/models.py:315]: expense_id

File: backend/api/models.py [hunk 6]
```diff
@@ -214,14 +237,16 @@ def save(self, *args, **kwargs):
 
 
 class OtherPlanningData(models.Model):
```
Comment (edward-stalley-work) [inline backend/api/models.py:239]: If table is being used. Need to discuss

class OtherPlanningData(models.Model):
→ class Analyses(models.Model):

File: backend/api/models.py [hunk 7]
```diff
@@ -214,14 +237,16 @@ def save(self, *args, **kwargs):
...
+    analysis_id = models.CharField(max_length=10, primary_key=True, editable=False)
+    project_id = models.ForeignKey(
         PlanningProjectData, on_delete=models.CASCADE, related_name="other_planning"
     )
     gross_profit = models.IntegerField(max_length=12)
     net_profit_for_the_period = models.IntegerField(max_length=12)
     gross_profit_margin = models.FloatField(max_length=10)
     operating_profit_margin = models.FloatField(max_length=10)
+    created_At = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now_add=True)
```
Comment (edward-stalley-work) [inline backend/api/models.py:249]: Do you need to add this temporarily before the class name changes?

 class Meta :
        db_table = u'analyses'

File: backend/api/models.py (hunk 6 above)
Comment (edward-stalley-work) [inline backend/api/models.py:239]: Ishigamori san had a note saying this table can be removed if not being used.


I named it analyses - but I know that the 'planning profit and loss' screen and the 'analysis' screen just use existing data so they do not really need their own DB Table.

Currently - What exactly is this table "otherplanning" table for? Can it be removed?
Please reply to me here in regard to this matter.

**Review by edward-stalley-work — CHANGES_REQUESTED** [review]
Thanks for your hard work Merlou.
I have left some comments.
Please look at them and either reply with - 'fixed'   or your thoughts/replies.

File: backend/api/models.py [hunk 8]
```diff
@@ -1,37 +1,19 @@
...
-        return f"Account Master: ID={self.id}, Sales Revenue={self.sales_revenue}"
-
-
 class ClientMaster(models.Model):
     client_id = models.CharField(max_length=10, primary_key=True, editable=False)
     client_name = models.CharField(max_length=100)
     created_at = models.DateTimeField(auto_now_add=True)
-    registered_user_id = models.CharField(max_length=100)
+    auth_user_id = models.ForeignKey(AuthUser, on_delete=models.CASCADE)
+    updated_at = models.DateTimeField(auto_now_add=True)
```
[the remark this answers is not in the document]
Comment (merlouduron) [reply backend/api/models.py:14]: Fixed.

File: backend/api/models.py [hunk 9]
```diff
@@ -100,15 +88,19 @@ def save(self, *args, **kwargs):
 
 
 class User(AbstractBaseUser):
-    user_id = models.CharField(max_length=10, primary_key=True, editable=False)
-    username = models.TextField()
-    email = models.EmailField()
-    affiliated_company_id = models.ForeignKey(CompanyMaster, on_delete=models.CASCADE)
-    affiliated_business_division_id = models.ForeignKey(
+    employee_id = models.CharField(max_length=10, primary_key=True, editable=False)
+    firstname = models.CharField(max_length=100)
```
[the remark this answers is not in the document]
Comment (merlouduron) [reply backend/api/models.py]: Fixed.

File: backend/api/models.py [hunk 10]
```diff
@@ -100,15 +88,19 @@ def save(self, *args, **kwargs):
...
+    lastname = models.CharField(max_length=100)
```
[the remark this answers is not in the document]
Comment (merlouduron) [reply backend/api/models.py]: Fixed

File: backend/api/models.py [hunk 11]
```diff
@@ -100,15 +88,19 @@ def save(self, *args, **kwargs):
 
 
 class User(AbstractBaseUser):
-    user_id = models.CharField(max_length=10, primary_key=True, editable=False)
-    username = models.TextField()
-    email = models.EmailField()
```
[the remark this answers is not in the document]
Comment (merlouduron) [reply backend/api/models.py:105]: Fixed.

File: backend/api/models.py (hunk 1 above)
Comment (merlouduron) [reply backend/api/models.py:122]: This will be updated in Ticket 73.

File: backend/api/models.py (hunk 2 above)
Comment (merlouduron) [reply backend/api/models.py:130]: Fixed.

File: backend/api/models.py (hunk 3 above)
Comment (merlouduron) [reply backend/api/models.py:131]: I don't know which part would be removed.

File: backend/api/models.py (hunk 4 above)
Comment (merlouduron) [reply backend/api/models.py]: Fixed.

File: backend/api/models.py (hunk 5 above)
Comment (merlouduron) [reply backend/api/models.py]: Fixed.

File: backend/api/models.py (hunk 6 above)
Comment (merlouduron) [reply backend/api/models.py]: Table is removed.

Comment (merlouduron) [reply backend/api/models.py]: Table is removed.

File: backend/api/models.py (hunk 8 above)
[the remark this answers is not in the document]
Comment (edward-stalley-work) [reply backend/api/models.py:14]: Some of the models still have a different order.
Please doublecheck!

File: backend/api/models.py [hunk 12]
```diff
@@ -40,7 +40,7 @@ class CompanyMaster(models.Model):
     company_id = models.CharField(max_length=10, primary_key=True, editable=False)
     company_name = models.CharField(max_length=100)
     created_at = models.DateTimeField(auto_now_add=True)
-    auth_user_id = models.ForeignKey(AuthUser, on_delete=models.CASCADE)
+    auth_user = models.ForeignKey(AuthUser, on_delete=models.CASCADE)
```
Comment (edward-stalley-work) [inline backend/api/models.py:43]: move this above 'created_at'

auth_user , created_at, updated_at

File: backend/api/models.py [hunk 13]
```diff
@@ -89,15 +89,16 @@ def save(self, *args, **kwargs):
...
     salary = models.IntegerField(max_length=12)
-    company_id = models.ForeignKey(CompanyMaster, on_delete=models.CASCADE)
-    business_division_id = models.ForeignKey(
+    company = models.ForeignKey(CompanyMaster, on_delete=models.CASCADE)
+    business_division = models.ForeignKey(
         BusinessDivisionMaster, on_delete=models.CASCADE
     )
     created_at = models.DateTimeField(auto_now_add=True)
-    auth_user_id =  models.ForeignKey(AuthUser, on_delete=models.CASCADE)
+    auth_user =  models.ForeignKey(AuthUser, on_delete=models.CASCADE)
```
Comment (edward-stalley-work) [inline backend/api/models.py:101]: move this above 'created_at'

auth_user , created_at, updated_at

File: backend/api/models.py (hunk 3 above)
Comment (edward-stalley-work) [reply backend/api/models.py:131]: Sorry Merlou I do not understand what you mean?

Can this be removed from model? It does not seem to serve a purpose.

**Review by edward-stalley-work — CHANGES_REQUESTED** [review]
Please check comments!

File: backend/api/models.py (hunk 8 above)
[the remark this answers is not in the document]
Comment (merlouduron) [reply backend/api/models.py:14]: Fixed.

File: backend/api/models.py (hunk 12 above)
Comment (merlouduron) [reply backend/api/models.py:42]: Fixed.

File: backend/api/models.py (hunk 13 above)
Comment (merlouduron) [reply backend/api/models.py:100]: Fixed.

File: backend/api/models.py (hunk 3 above)
Comment (merlouduron) [reply backend/api/models.py:131]: Do you mean these two ?
project_name = models.CharField(max_length=100)
project_type = models.CharField(max_length=50, null=True)
If yes,  cause they serve purpose cause they're used for the display on a certain page. 
Correct me if I'm wrong. 🙇

Comment (edward-stalley-work) [reply backend/api/models.py:131]: No worries - I mean this
`  planning = models.DateField(null=True, default=timezone.now)`

Comment (merlouduron) [reply backend/api/models.py]: Done deleting the unused table column. 🙇 
Fixed.

**Pull request merged**